samcli/lib/build/app_builder.py

```diff
@@ -182,20 +183,44 @@ def _build_function(self, function_name, codeuri, runtime):
             if self._container_manager:
                 build_method = self._build_function_on_container
 
+            options = ApplicationBuilder._get_build_options(config.language, handler)
+
             return build_method(config,
                                 code_dir,
                                 artifacts_dir,
                                 scratch_dir,
                                 manifest_path,
-                                runtime)
+                                runtime,
+                                options)
+
+    @staticmethod
+    def _get_build_options(language, handler):
+        """
+
+        Parameters
+        ----------
+        language str
+            Language of the runtime
+        handler str
+            Handler value of the Lambda Function Resource
+        Returns
+        -------
+        dict
+            Dictionary that represents the options to pass to the builder workflow or None if options are not needed
+        """
+        options = None
```

Comment:
``
{'artifact_executable_name': handler} if language=='go' else None
``

could be simpler